io-engine/src/lvm/lv_replica.rs

```diff
@@ -0,0 +1,739 @@
+use super::{cli::de, error::Error, vg_pool::VolumeGroup, CmnQueryArgs};
+use crate::{
+    bdev::PtplFileOps,
+    bdev_api::{bdev_create, BdevError},
+    core::{Protocol, PtplProps, Share, ShareProps, UntypedBdev, UpdateProps},
+    lvm::{
+        cli::LvmCmd,
+        property::{Property, PropertyType},
+    },
+};
+
+use std::{
+    ops::{Deref, DerefMut},
+    pin::Pin,
+};
+
+/// Different list options for a logical volume.
+#[derive(Default, Debug)]
+pub(crate) struct QueryArgs {
+    /// Pertaining the volume group parent.
+    vg: super::vg_pool::QueryArgs,
+    /// Pertaining the logical volume itself.
+    lv: CmnQueryArgs,
+    /// Our lv's are "special", as in we set the uuid to the name, and add the
+    /// name as a lv_tag (LVM_TAG.lv.name=name)
+    regular_lv: bool,
+}
+impl QueryArgs {
+    /// Builder-like creating a default `Self`.
+    pub(crate) fn new() -> Self {
+        Self::default()
+    }
+    /// Add the VG query args.
+    pub(crate) fn with_vg(self, vg: CmnQueryArgs) -> Self {
+        Self {
+            vg: vg.into(),
+            ..self
+        }
+    }
+    /// Add the LV query args.
+    pub(crate) fn with_lv(self, lv: CmnQueryArgs) -> Self {
+        Self {
+            lv,
+            ..self
+        }
+    }
+    /// Get a comma-separated list of query selection args.
+    /// todo: should be Display trait?
+    pub(super) fn query(&self) -> String {
+        let mut select = self.vg.query();
+        let args = &self.lv;
+
+        if self.regular_lv {
+            if let Some(lv_name) = &args.name {
+                select.push_str(&format!("lv_name={lv_name},"));
+            }
+            if let Some(lv_uuid) = &args.uuid {
+                select.push_str(&format!("lv_uuid={lv_uuid},"));
+            }
+        } else {
+            if let Some(name) = &args.name {
+                select.push_str(&format!(
+                    "lv_tags={},",
+                    Property::LvName(name.to_string()).tag()
+                ));
+            }
+            if let Some(lv_name) = &args.uuid {
+                select.push_str(&format!("lv_name={lv_name},"));
+            }
+        }
+
+        if let Some(lv_tag) = &args.tag {
+            select.push_str(&format!("lv_tags={lv_tag},"));
+        }
+        select
+    }
+}
+
+/// A Logical Volume (LV) is a virtual block device that can be used by the
+/// system or applications.
+/// Each block of data in an LV is stored on one or more PV in the VG, according
+/// to algorithms implemented by Device Mapper (DM) in the kernel.
+#[derive(Debug, Clone, Deserialize)]
+pub struct LogicalVolume {
```

Comment:
Are we going to impl the `LogicalVolume` trait for this lvm `LogicalVolume` type, like we have for spdk_lvol ? I'm wondering about the naming confusion here

Reply:
I'm not sure, existing `LogicalVolume` trait has spdk specific things, so I'm not sure how to integrate it...
Also the trait seems to be used only inside Lvs so far.